libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -102,6 +119,41 @@ func (c *Config) Validate() error {
 	if c.Scope != "node" && c.Scope != "cluster" {
 		return fmt.Errorf("invalid `scope` configured. supported values are `node` and `cluster`")
 	}
+
+	if c.Sharding.Count != 0 {
+		if c.Scope == "node" {
+			logp.L().Warnf("can not set sharding.count to `%d` when scope to `cluster`", c.Sharding.Count)
```

Comment:
yes. i intentionally left documentation out of the first review so that i dont have to rewrite the whole thing if you guys have better ideas than this PR :)

the idea is that this feature shouldnt be allowed on scope=node. given the amount of resource fragmentation already present, adding this feature in at the node level can make things worse.

Reply:
```suggestion
			logp.L().Warnf("can not set sharding.count to `%d` when scope is set to `node`", c.Sharding.Count)
```